new packages (py-metaphlan, py-hclust2, iqtree2) + updates to others (py-dendropy, py-phylophlan, py-pkgconfig)

luke-dt opened the pull request:
I added a metaphlan package, and in the process cleaned up the py-phylophlan package with the dependencies from the bioconda recipe, added some newer versions of dependencies, and fixed py-pkgconfig which was throwing an error while building

luke-dt committed:
added metaphlan v4, cleaned up phylophlan

luke-dt committed:
added iqtree2

luke-dt committed:
fixed phylophlan, builds now

luke-dt committed:
changed config.yaml to default

luke-dt committed:
fixed style

luke-dt committed:
Merge branch 'develop' into package/py-metaphlan

adamjstewart commented on var/spack/repos/builtin/packages/iqtree2/package.py:
```diff
@@ -0,0 +1,20 @@
+
+    version("2.1.2", sha256="3aaf5ac7f60d852ac8b733fb82832c049ca48b7203a6a865e99c5af359fcca5a")
+
+    depends_on("boost", type=("build", "run"), when="@2:")
+    depends_on("eigen", type=("build", "run"), when="@1.6:")
+    depends_on("zlib", type=("build", "run"))
```
I highly doubt any of these dep types are correct. Build/run means to put `prefix.bin` in `PATH`, but none of these even come with executables.

luke-dt replied:
so should they just be build dependencies?

adamjstewart replied:
They should just be link dependencies if there are no exes

adamjstewart commented on var/spack/repos/builtin/packages/py-dendropy/package.py:
```diff
@@ -16,6 +16,7 @@ class PyDendropy(PythonPackage):
+    version("4.5.2", sha256="3e5d2522170058ebc8d1ee63a7f2d25b915e34957dc02693ebfdc15f347a0101")
     version("4.3.0", sha256="bd5b35ce1a1c9253209b7b5f3939ac22beaa70e787f8129149b4f7ffe865d510")
     version("3.12.0", sha256="38a0f36f2f7aae43ec5599408b0d0a4c80996b749589f025940d955a70fc82d4")
 
```
setuptools should be build/run

adamjstewart commented on var/spack/repos/builtin/packages/py-metaphlan/package.py:
```diff
@@ -0,0 +1,41 @@
+
+    depends_on("python@3.7:", type=("build", "run"))
+    depends_on("py-setuptools", type="build")
+    depends_on("py-cython", type=("build", "run"))
```
cython doesn't seem to be used by this package

luke-dt replied:
It was in the bioconda recipe, so I copied it

adamjstewart replied:
I don't know what bioconda is, and I definitely don't trust it

luke-dt replied:
Bioconda is the most popular conda channel for bioinformatics, and the software authors are the ones that manage the bioconda recipe, so I have no reason to doubt it.

luke-dt replied:
So cython is listed as a "host dependency" in the conda recipe, which I am unfamiliar with. Apparently they are "packages that need to be specific to the target platform when the target platform is not necessarily the same as the native build platform"

https://docs.conda.io/projects/conda-build/en/latest/resources/define-metadata.html#host

What would the equivalent type be for spack/is it necessary to declare it for spack?

adamjstewart replied:
When you build the package, is it actually using cython anywhere? Can you upload the build log? There's no mention of cython anywhere in the source code.

luke-dt replied:
This log was from a build on a different system, but I don't see it mentioned anywhere

[spack-build-out.txt](https://github.com/spack/spack/files/9693148/spack-build-out.txt)

adamjstewart replied:
Yep, seems like it's not needed, which makes sense because it isn't used by the build system.

adamjstewart commented on var/spack/repos/builtin/packages/py-metaphlan/package.py:
```diff
@@ -0,0 +1,41 @@
+    version("4.0.2", sha256="2549fdf2de97a0024551a7bb8d639613b8a7b612054506c88cdb719353f466ff")
+    version("3.1.0", sha256="4e7a7a36d07ed6f4f945afc4216db7f691d44a22b059c2404c917a160a687a6b")
+
+    depends_on("python@3.7:", type=("build", "run"))
```
Where does 3.7 come from?

luke-dt replied:
https://github.com/bioconda/bioconda-recipes/blob/master/recipes/metaphlan/meta.yaml

adamjstewart commented on var/spack/repos/builtin/packages/py-phylophlan/package.py:
```diff
@@ -11,17 +12,25 @@ class PyPhylophlan(PythonPackage):
-    depends_on("py-matplotlib", type=("build", "run"))
-    depends_on("py-pandas", type=("build", "run"))
-    depends_on("py-seaborn", type=("build", "run"))
+    depends_on("python@3.7:", type=("build", "run"))
```
Where does 3.7 come from?

luke-dt replied:
https://github.com/bioconda/bioconda-recipes/blob/master/recipes/phylophlan/meta.yaml

adamjstewart commented on var/spack/repos/builtin/packages/py-phylophlan/package.py:
```diff
@@ -11,17 +12,25 @@ class PyPhylophlan(PythonPackage):
+    depends_on("muscle@3.8.1551:", type=("build", "run"))
+    depends_on("mafft@7.310:", type=("build", "run"))
+    depends_on("fasttree@2.1.8:", type=("build", "run"))
+    depends_on("raxml@8.2.10:", type=("build", "run"))
```
Where do all these dependency versions come from?

luke-dt replied:
https://github.com/bioconda/bioconda-recipes/blob/master/recipes/phylophlan/meta.yaml

luke-dt replied:
The wiki just lists the tools (https://github.com/biobakery/phylophlan/wiki#external-tools) but for some reason the bioconda recipe is WAY more specific with versions.

adamjstewart commented on var/spack/repos/builtin/packages/py-pkgconfig/package.py:
```diff
@@ -20,7 +20,7 @@ class PyPkgconfig(PythonPackage):
-    depends_on("py-poetry-core@1:", when="@1.5.5:", type="build")
-    depends_on("py-poetry-core@0.12:", when="@1.5:", type="build")
+    depends_on("py-poetry@1:", when="@1.5.2:", type="build")
+    depends_on("py-poetry@0.12:", when="@1.5:", type="build")
```
This is incorrect

luke-dt replied:
I kept getting "No module named poetry" error while installing this, but I'll test it again without this

luke-dt replied:
@adamjstewart I'm having the same issue when re-building. I think it's a similar problem to https://github.com/spack/spack/issues/29308 Here is the build log:

```
==> py-pkgconfig: Executing phase: 'install'
==> [2022-10-01-10:45:23.400953] '/home/lukedt/Documents/GitHub/spack/opt/spack/linux-fedora36-zen/gcc-8.5.0/python-3.9.13-5zn3hui4ysxknqhfvmbsupmf5ysefebn/bin/python3.9' '-m' 'pip' '-vvv' '--no-input' '--no-cache-dir' '--disable-pip-version-check' 'install' '--no-deps' '--ignore-installed' '--no-build-isolation' '--no-warn-script-location' '--no-index' '--prefix=/home/lukedt/Documents/GitHub/spack/opt/spack/linux-fedora36-zen/gcc-8.5.0/py-pkgconfig-1.5.5-yzezoqcuwxhqkkc4z6o7purymq3ufhe6' '.'i.org/simple
Created temporary directory: /tmp/pip-ephem-wheel-cache-v265qa90
Created temporary directory: /tmp/pip-build-tracker-rrrnkivo
Initialized build tracking at /tmp/pip-build-tracker-rrrnkivo
Created build tracker: /tmp/pip-build-tracker-rrrnkivo
Entered build tracker: /tmp/pip-build-tracker-rrrnkivo
Created temporary directory: /tmp/pip-install-jbh6ib2n
Processing /home/lukedt/Documents/GitHub/spack/var/spack/stage/spack-stage-py-pkgconfig-1.5.5-yzezoqcuwxhqkkc4z6o7purymq3ufhe6/spack-src
  Added file:///home/lukedt/Documents/GitHub/spack/var/spack/stage/spack-stage-py-pkgconfig-1.5.5-yzezoqcuwxhqkkc4z6o7purymq3ufhe6/spack-src to build tracker '/tmp/pip-build-tracker-rrrnkivo'
  Created temporary directory: /tmp/pip-modern-metadata-ge9ca6vp
  Preparing metadata (pyproject.toml): started
  Running command Preparing metadata (pyproject.toml)
  Preparing metadata (pyproject.toml): finished with status 'done'
ERROR: Exception:
Traceback (most recent call last):
  File "/home/lukedt/Documents/GitHub/spack/opt/spack/linux-fedora36-zen/gcc-8.5.0/py-pip-22.1.2-frngqanail6q44luijakcnmb3iox3fws/lib/python3.9/site-packages/pip/_internal/cli/base_command.py", line 167, in exc_logging_wrapper
    status = run_func(*args)
  File "/home/lukedt/Documents/GitHub/spack/opt/spack/linux-fedora36-zen/gcc-8.5.0/py-pip-22.1.2-frngqanail6q44luijakcnmb3iox3fws/lib/python3.9/site-packages/pip/_internal/cli/req_command.py", line 205, in wrapper
    return func(self, options, args)
  File "/home/lukedt/Documents/GitHub/spack/opt/spack/linux-fedora36-zen/gcc-8.5.0/py-pip-22.1.2-frngqanail6q44luijakcnmb3iox3fws/lib/python3.9/site-packages/pip/_internal/commands/install.py", line 341, in run
    requirement_set = resolver.resolve(
  File "/home/lukedt/Documents/GitHub/spack/opt/spack/linux-fedora36-zen/gcc-8.5.0/py-pip-22.1.2-frngqanail6q44luijakcnmb3iox3fws/lib/python3.9/site-packages/pip/_internal/resolution/resolvelib/resolver.py", line 75, in resolve
    collected = self.factory.collect_root_requirements(root_reqs)
  File "/home/lukedt/Documents/GitHub/spack/opt/spack/linux-fedora36-zen/gcc-8.5.0/py-pip-22.1.2-frngqanail6q44luijakcnmb3iox3fws/lib/python3.9/site-packages/pip/_internal/resolution/resolvelib/factory.py", line 506, in collect_root_requirements
    req = self._make_requirement_from_install_req(
  File "/home/lukedt/Documents/GitHub/spack/opt/spack/linux-fedora36-zen/gcc-8.5.0/py-pip-22.1.2-frngqanail6q44luijakcnmb3iox3fws/lib/python3.9/site-packages/pip/_internal/resolution/resolvelib/factory.py", line 468, in _make_requirement_from_install_req
    cand = self._make_candidate_from_link(
  File "/home/lukedt/Documents/GitHub/spack/opt/spack/linux-fedora36-zen/gcc-8.5.0/py-pip-22.1.2-frngqanail6q44luijakcnmb3iox3fws/lib/python3.9/site-packages/pip/_internal/resolution/resolvelib/factory.py", line 215, in _make_candidate_from_link
    self._link_candidate_cache[link] = LinkCandidate(
  File "/home/lukedt/Documents/GitHub/spack/opt/spack/linux-fedora36-zen/gcc-8.5.0/py-pip-22.1.2-frngqanail6q44luijakcnmb3iox3fws/lib/python3.9/site-packages/pip/_internal/resolution/resolvelib/candidates.py", line 291, in __init__
    super().__init__(
  File "/home/lukedt/Documents/GitHub/spack/opt/spack/linux-fedora36-zen/gcc-8.5.0/py-pip-22.1.2-frngqanail6q44luijakcnmb3iox3fws/lib/python3.9/site-packages/pip/_internal/resolution/resolvelib/candidates.py", line 161, in __init__
    self.dist = self._prepare()
  File "/home/lukedt/Documents/GitHub/spack/opt/spack/linux-fedora36-zen/gcc-8.5.0/py-pip-22.1.2-frngqanail6q44luijakcnmb3iox3fws/lib/python3.9/site-packages/pip/_internal/resolution/resolvelib/candidates.py", line 230, in _prepare
    dist = self._prepare_distribution()
  File "/home/lukedt/Documents/GitHub/spack/opt/spack/linux-fedora36-zen/gcc-8.5.0/py-pip-22.1.2-frngqanail6q44luijakcnmb3iox3fws/lib/python3.9/site-packages/pip/_internal/resolution/resolvelib/candidates.py", line 302, in _prepare_distribution
    return preparer.prepare_linked_requirement(self._ireq, parallel_builds=True)
  File "/home/lukedt/Documents/GitHub/spack/opt/spack/linux-fedora36-zen/gcc-8.5.0/py-pip-22.1.2-frngqanail6q44luijakcnmb3iox3fws/lib/python3.9/site-packages/pip/_internal/operations/prepare.py", line 428, in prepare_linked_requirement
    return self._prepare_linked_requirement(req, parallel_builds)
  File "/home/lukedt/Documents/GitHub/spack/opt/spack/linux-fedora36-zen/gcc-8.5.0/py-pip-22.1.2-frngqanail6q44luijakcnmb3iox3fws/lib/python3.9/site-packages/pip/_internal/operations/prepare.py", line 497, in _prepare_linked_requirement
    dist = _get_prepared_distribution(
  File "/home/lukedt/Documents/GitHub/spack/opt/spack/linux-fedora36-zen/gcc-8.5.0/py-pip-22.1.2-frngqanail6q44luijakcnmb3iox3fws/lib/python3.9/site-packages/pip/_internal/operations/prepare.py", line 58, in _get_prepared_distribution
    abstract_dist.prepare_distribution_metadata(
  File "/home/lukedt/Documents/GitHub/spack/opt/spack/linux-fedora36-zen/gcc-8.5.0/py-pip-22.1.2-frngqanail6q44luijakcnmb3iox3fws/lib/python3.9/site-packages/pip/_internal/distributions/sdist.py", line 61, in prepare_distribution_metadata
    self.req.prepare_metadata()
  File "/home/lukedt/Documents/GitHub/spack/opt/spack/linux-fedora36-zen/gcc-8.5.0/py-pip-22.1.2-frngqanail6q44luijakcnmb3iox3fws/lib/python3.9/site-packages/pip/_internal/req/req_install.py", line 526, in prepare_metadata
    self.metadata_directory = generate_metadata(
  File "/home/lukedt/Documents/GitHub/spack/opt/spack/linux-fedora36-zen/gcc-8.5.0/py-pip-22.1.2-frngqanail6q44luijakcnmb3iox3fws/lib/python3.9/site-packages/pip/_internal/operations/build/metadata.py", line 35, in generate_metadata
    distinfo_dir = backend.prepare_metadata_for_build_wheel(metadata_dir)
  File "/home/lukedt/Documents/GitHub/spack/opt/spack/linux-fedora36-zen/gcc-8.5.0/py-pip-22.1.2-frngqanail6q44luijakcnmb3iox3fws/lib/python3.9/site-packages/pip/_internal/utils/misc.py", line 706, in prepare_metadata_for_build_wheel
    return super().prepare_metadata_for_build_wheel(
  File "/home/lukedt/Documents/GitHub/spack/opt/spack/linux-fedora36-zen/gcc-8.5.0/py-pip-22.1.2-frngqanail6q44luijakcnmb3iox3fws/lib/python3.9/site-packages/pip/_vendor/pep517/wrappers.py", line 188, in prepare_metadata_for_build_wheel
    return self._call_hook('prepare_metadata_for_build_wheel', {
  File "/home/lukedt/Documents/GitHub/spack/opt/spack/linux-fedora36-zen/gcc-8.5.0/py-pip-22.1.2-frngqanail6q44luijakcnmb3iox3fws/lib/python3.9/site-packages/pip/_vendor/pep517/wrappers.py", line 332, in _call_hook
    raise BackendUnavailable(data.get('traceback', ''))
pip._vendor.pep517.wrappers.BackendUnavailable: Traceback (most recent call last):
  File "/home/lukedt/Documents/GitHub/spack/opt/spack/linux-fedora36-zen/gcc-8.5.0/py-pip-22.1.2-frngqanail6q44luijakcnmb3iox3fws/lib/python3.9/site-packages/pip/_vendor/pep517/in_process/_in_process.py", line 89, in _build_backend
    obj = import_module(mod_path)
  File "/home/lukedt/Documents/GitHub/spack/opt/spack/linux-fedora36-zen/gcc-8.5.0/python-3.9.13-5zn3hui4ysxknqhfvmbsupmf5ysefebn/lib/python3.9/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1030, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1007, in _find_and_load
  File "<frozen importlib._bootstrap>", line 972, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 228, in _call_with_frames_removed
  File "<frozen importlib._bootstrap>", line 1030, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1007, in _find_and_load
  File "<frozen importlib._bootstrap>", line 972, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 228, in _call_with_frames_removed
  File "<frozen importlib._bootstrap>", line 1030, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1007, in _find_and_load
  File "<frozen importlib._bootstrap>", line 972, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 228, in _call_with_frames_removed
  File "<frozen importlib._bootstrap>", line 1030, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1007, in _find_and_load
  File "<frozen importlib._bootstrap>", line 984, in _find_and_load_unlocked
ModuleNotFoundError: No module named 'poetry'

Removed file:///home/lukedt/Documents/GitHub/spack/var/spack/stage/spack-stage-py-pkgconfig-1.5.5-yzezoqcuwxhqkkc4z6o7purymq3ufhe6/spack-src from build tracker '/tmp/pip-build-tracker-rrrnkivo'
Removed build tracker: '/tmp/pip-build-tracker-rrrnkivo'
```

adamjstewart replied:
Can you share `spack-built-env.txt` and also the output of `python3 -m sysconfig`?

adamjstewart replied:
Are you using Spack-installed Python or the system Python as an external package?

luke-dt replied:
This system has python installed separately, yes.

[spack-build-env.txt](https://github.com/spack/spack/files/9693183/spack-build-env.txt)
[python3_sysconfig.txt](https://github.com/spack/spack/files/9693186/python3_sysconfig.txt)

adamjstewart replied:
Okay, so you're using Spack-installed Python. So it's not an issue with sysconfig/distutils hacking.

Another possibility is that this is related to #32375. Can you run:
```console
$ tree /home/lukedt/Documents/GitHub/spack/opt/spack/linux-fedora36-zen/gcc-8.5.0/py-poetry-core-1.2.0-xr4smvmirljhej6hnwtnsb5bmlrnyg6b/lib/python3.9/site-packages
```
(or `find` if `tree` is not installed)

I'm guessing that the entire `poetry` directory and all of its contents will be missing. Do you have a global `.gitignore` somewhere? Can you upload that?

luke-dt replied:
```
/home/lukedt/Documents/GitHub/spack/opt/spack/linux-fedora36-zen/gcc-8.5.0/py-poetry-core-1.2.0-xr4smvmirljhej6hnwtnsb5bmlrnyg6b/lib/python3.9/site-packages
└── poetry_core-1.2.0.dist-info
    ├── direct_url.json
    ├── INSTALLER
    ├── LICENSE
    ├── METADATA
    ├── RECORD
    ├── REQUESTED
    └── WHEEL

1 directory, 7 files

```
I don't have a global `.gitignore`, just the one in `~/Documents/GitHub/spack`

adamjstewart replied:
Must be pulling it in from somewhere else then. I'll see if I can hack together a solution since you're the 3rd person who's reported this issue.

luke-dt replied:
thanks for the assistance!

luke-dt replied:
so oddly enough, my latest commit ([e64fd01](https://github.com/spack/spack/pull/32936/commits/e64fd01bbf0c76b5aa145751ec999e167eafefa2)) builds OK on my machine without this error...

manuelakuhn committed:
py-jsonschema: add 4.16.0 and new package py-hatch-fancy-pypi-readme (#32929)

annop-w committed:
acfl: add v22.1 (#32915)

Co-authored-by: Annop Wongwathanarat <annop.wongwathanarat@arm.com>

johnwparent committed:
Fixup errors introduced by Clingo Pr: (#32905)

* re2c depends on cmake on Windows
* Winbison properly added to bootstrap package search list

aurianer committed:
Set CMAKE_HIP_ARCHITECTURES with the value of amdgpu_target (#32901)

adamjstewart committed:
libtiff: default to +zlib+jpeg (#32945)

siko1056 committed:
octave: add version 7.2.0 (#32943)

viniciusvgp committed:
simgrid new releases (#32920)

mdorier committed:
[rocksdb] Added rtti variant (#32918)

renjithravindrankannath committed:
rvs binary path updated for 5.2 rocm release (#32892)

iarspider committed:
Add checksum for py-pytest-runner 6.0.0 (#32957)

adamjstewart committed:
py-einops: add v0.5.0 (#32959)

aurianer committed:
Replace repo with the NVIDIA one (#32951)

iarspider committed:
Add checksum for py-tomli 2.0.1 (#32949)

prckent committed:
QMCPACK: add @3.15.0 (#32931)

bvanessen committed:
Tidied up configure arguments to use special spack autotools features. (#32930)

snehring committed:
casper: old domain fell off, adding github repo (#32928)

CamStan committed:
unifyfs: pin mercury version; add boost variant (#32911)

Mercury has a new version (v2.2) releasing soon that UnifyFS does not build with and hasn't been tested with. This pins UnifyFS to the last version of Mercury used/tested.

Add a variant to avoid building/using boost

Append -std=gnu99 to cflags if building with gcc@4. Needed for mochi-margo to compile

balos1 committed:
trilinos: constrain superlu-dist version (#32889)

* trilinos: constrain superlu-dist version for 13.x
* syntax

jhale committed:
FEniCSx: Updates for 0.5.1 (#32665)

* Updates for DOLFINx 0.5.1 and associated packages
* xtensor needed on anything less than main
* Switch back to Python 3.7 minimum.
* Might be good to point out in our README how to fix Python version?
* Fix basix, xtensor dep
* Add numba feature
* Fix checksum
* Make slepc optional

Co-authored-by: Tamara Dahlgren <35777542+tldahlgren@users.noreply.github.com>

llnns committed:
simgrid: add variant and remove flag (#32797)

* simgrid: remove std c++11 flag
* simgrid: add msg variant

white238 committed:
Axom: bring in changes from axom repo (#32643)

* bring in changes from axom repo

Co-authored-by: white238 <white238@users.noreply.github.com>
Co-authored-by: Tamara Dahlgren <35777542+tldahlgren@users.noreply.github.com>

iarspider committed:
Add checksum for py-pyparsing 3.0.9 (#32952)

mpokorny committed:
rdma-core: fix syntax for external discoverability (#32962)

iarspider committed:
Add checksum for py-flatbuffers 2.0.7 (#32955)

WeiqunZhang committed:
amrex: add v22.10 (#32966)

alalazo committed:
Remove CMakePackage.define alias from most packages (#32950)

climbfuji committed:
Bug fix for `ca-certificates-mozilla/package.py` to enable `spack install --source` (#32953)

adamjstewart commented:
Looks like a bad rebase...

luke-dt commented:
> Looks like a bad rebase...

Yeah, I was trying to fetch the upstream changes and test on a different machine and messed that up.

luke-dt committed:
Merge branch 'develop' into package/py-metaphlan

luke-dt committed:
made suggested changes to iqtree2, py-dendropy, py-metaphlan, and py-pkgconfig. Poetry install still broken

luke-dt committed:
reverted py-pkgconfig deps to poetry-core

adamjstewart commented on var/spack/repos/builtin/packages/iqtree2/package.py:
```diff
@@ -0,0 +1,20 @@
+
+    version("2.1.2", sha256="3aaf5ac7f60d852ac8b733fb82832c049ca48b7203a6a865e99c5af359fcca5a")
+
+    depends_on("boost", type="link", when="@2:")
+    depends_on("eigen", type="link", when="@1.6:")
```
You only have 1 version so you could probably drop the `when` here

luke-dt replied:
true, and the iqtree2 package is by definition >1.6

adamjstewart commented on var/spack/repos/builtin/packages/py-dendropy/package.py:
```diff
@@ -16,8 +16,9 @@ class PyDendropy(PythonPackage):
     version("4.3.0", sha256="bd5b35ce1a1c9253209b7b5f3939ac22beaa70e787f8129149b4f7ffe865d510")
     version("3.12.0", sha256="38a0f36f2f7aae43ec5599408b0d0a4c80996b749589f025940d955a70fc82d4")
 
     depends_on("python@2.7:,3.4:")
```
```suggestion
    depends_on("python@2.7:,3.4:", type=("build", "run"))
```

adamjstewart commented on var/spack/repos/builtin/packages/py-pkgconfig/package.py:
```diff
@@ -20,7 +20,7 @@ class PyPkgconfig(PythonPackage):
     depends_on("python@2.7,3.3:", when="@1.5:", type=("build", "run"))
     depends_on("python@2.6:2.7,3.3:", when="@1.4:", type=("build", "run"))
-    depends_on("py-poetry-core@1:", when="@1.5.5:", type="build")
+    depends_on("py-poetry-core@1:", when="@1.5.2:", type="build")
     depends_on("py-poetry-core@0.12:", when="@1.5:", type="build")
```
Just took another look at this and it looks like it should actually be:
```suggestion
    depends_on("py-poetry-core@1:", when="@1.5.2:", type="build")
    depends_on("py-poetry@0.12:", when="@1.5.0:1.5.1", type="build")
```

luke-dt committed:
made iqtree2 less dedundant, changes to py-dendropy and py-pkgconfig deps

adamjstewart commented:
@spackbot run pipeline

luke-dt commented:
What check is it failing? I don't really understand what the pipeline is testing.

adamjstewart commented:
This is a false positive. I'll try rebasing to see if that solves the issue. If not, I'll ping some of our CI folks.

adamjstewart commented:
It looks like I don't have permission to rebase. Can you either rebase or merge from develop to update your branch?

luke-dt committed:
Merge branch 'develop' into package/py-metaphlan
